filter/src/main/java/gov/cms/ab2d/filter/ExplanationOfBenefitTrimmerR4.java

```diff
@@ -2,108 +2,201 @@
+ *     . insurer
+ *     . insurerTarget
+ *     . provider
+ *     . providerTarget
+ *     . referral
+ *     . referralTarget
+ *     . facilityTarget
+ *     . claim
+ *     . claimTarget
+ *     . claimResponse
+ *     . claimResponseTarget
+ *     . outcome
+ *     . disposition
+ *     . related
+ *     . prescription
+ *     . prescriptionTarget
+ *     . originalPrescription
+ *     . originalPrescriptionTarget
+ *     . payee
+ *     . precedence
+ *     . insurance
+ *     . accident
+ *     . supportingInfo (was information in STU3)
+ *     . addItem
+ *     . payment
+ *     . form
+ *     . contained
+ *     . processNote
+ *     . benefitBalance
+ *     . priority (new for R4)
+ *     . total (new for R4)
+ *     . use (new for R4)
+ *     . fundsReserveRequested (new for R4)
+ *     . fundsReserve (new for R4)
+ *     . preAuthRef (new for R4)
+ *     . preAuthRefPeriod (new for R4)
+ *     . formCode (new for R4)
+ *     . benefitPeriod (new for R4)
+ *     . adjucation (new for R4)
+ *
+ *  For item elements:
+ *
+ *  Keep:
+ *     . sequence
+ *     . careTeamSequence
+ *     . productOrService
+ *     . serviced
+ *     . location
+ *     . quantity
+ *     . extension
+ *
+ *  Items not kept:
+ *     . diagnosisSequence
+ *     . procedureSequence
+ *     . informationSequence
+ *     . extension
+ *     . revenue
+ *     . category
+ *     . modifier
+ *     . programCode
+ *     . unitPrice
+ *     . factor
+ *     . net
+ *     . udi
+ *     . udiTarget
+ *     . bodySite
+ *     . subSite
+ *     . encounter
+ *     . encounterTarget
+ *     . noteNumber
+ *     . adjudication
+ *     . detail
+ *     . modifierExtension
+ *
  */
 public class ExplanationOfBenefitTrimmerR4 {
-
     /**
      * Pass in an ExplanationOfBenefit, return the copy without the data
      *
-     * @param b - the original ExplanationOfBenefit
+     * @param resource - the original ExplanationOfBenefit
      * @return the cleaned up copy
      */
-    public static IBaseResource getBenefit(IBaseResource b) {
-        ExplanationOfBenefit benefit = (ExplanationOfBenefit) b;
+    public static IBaseResource getBenefit(IBaseResource resource) {
+        ExplanationOfBenefit benefit = (ExplanationOfBenefit) resource;
         if (benefit == null) {
             return null;
         }
         // Copy it so we don't destroy the original
-        ExplanationOfBenefit newBenefit = benefit.copy();
+        ExplanationOfBenefit newBenefit = copyData(benefit);
         // Remove the unauthorized data
         cleanOutUnNeededData(newBenefit);
         // Return the sanitized data
         return newBenefit;
     }
 
+    /**
+     * Copy data from the old EOB to the new EOB. Had to do it piecemeal because
+     * ExplanationOfBenefit.copy() resulted in a stack overvlow
+     *
+     * @param benefit - the EOB to copy necessary data
+     * @return the new object
+     */
+    private static ExplanationOfBenefit copyData(ExplanationOfBenefit benefit) {
```

Comment:
Do we have tests that confirm this is equivalent to the old method? Like the two objects serialize to the same value. I am worried about a regression in here.

Reply:
This is for R4. There are no regressions yet. This will spell out in the e2e tests when we implement the v2 endpoints.